k9mail/src/main/java/com/fsck/k9/controller/MessagingController.java

```diff
@@ -130,79 +128,38 @@
     private static final String PENDING_COMMAND_MARK_ALL_AS_READ = "com.fsck.k9.MessagingController.markAllAsRead";
     private static final String PENDING_COMMAND_EXPUNGE = "com.fsck.k9.MessagingController.expunge";
 
-    private static MessagingController inst = null;
-    private BlockingQueue<Command> mCommands = new PriorityBlockingQueue<>();
-
-    private Thread mThread;
-    private Set<MessagingListener> mListeners = new CopyOnWriteArraySet<>();
-
-    private final ConcurrentHashMap<String, AtomicInteger> sendCount = new ConcurrentHashMap<>();
-
-    ConcurrentHashMap<Account, Pusher> pushers = new ConcurrentHashMap<>();
-
-    private final ExecutorService threadPool = Executors.newCachedThreadPool();
 
-    private MessagingListener checkMailListener = null;
+    private static MessagingController inst = null;
 
-    private final MemorizingMessagingListener memorizingMessagingListener = new MemorizingMessagingListener();
 
+    // injected state
     private final Context context;
-    private final NotificationController notificationController;
     private final Contacts contacts;
-    private volatile boolean stopped = false;
-
-    private static final Set<Flag> SYNC_FLAGS = EnumSet.of(Flag.SEEN, Flag.FLAGGED, Flag.ANSWERED, Flag.FORWARDED);
-
-    private void suppressMessages(Account account, List<LocalMessage> messages) {
-        EmailProviderCache cache = EmailProviderCache.getCache(account.getUuid(), context);
-        cache.hideMessages(messages);
-    }
-
-    private void unsuppressMessages(Account account, List<? extends Message> messages) {
-        EmailProviderCache cache = EmailProviderCache.getCache(account.getUuid(), context);
-        cache.unhideMessages(messages);
-    }
-
-    private boolean isMessageSuppressed(LocalMessage message) {
-        long messageId = message.getId();
-        long folderId = message.getFolder().getId();
-
-        EmailProviderCache cache = EmailProviderCache.getCache(message.getFolder().getAccountUuid(), context);
-        return cache.isMessageHidden(messageId, folderId);
-    }
-
-    private void setFlagInCache(final Account account, final List<Long> messageIds,
-            final Flag flag, final boolean newState) {
+    private final NotificationController notificationController;
 
-        EmailProviderCache cache = EmailProviderCache.getCache(account.getUuid(), context);
-        String columnName = LocalStore.getColumnNameForFlag(flag);
-        String value = Integer.toString((newState) ? 1 : 0);
-        cache.setValueForMessages(messageIds, columnName, value);
-    }
+    // private immutable state
```

Comment:
the immutable was referring to just the thread, the others are private mutable state I just forgot to annotate :)